Upgrade to Rebar3, additions to publish to Hex.pm

waisbrot opened the pull request:
Transferred from accense/eini#3 I believe that a new Hex release after this should fix erlcloud/erlcloud#307

Uses rebar3 as the build tool, but with a rebar.config.script for
rebar2 backwards-compatibility.

A few bits filled in on the .app.src file for Hex.pm publishing.

waisbrot committed:
Upgrade to Rebar3, additions to publish to Hex.pm

Uses rebar3 as the build tool, but with a rebar.config.script for
rebar2 backwards-compatibility.

A few bits filled in on the .app.src file for Hex.pm publishing.

motobob commented on rebar.config.script:
```diff
@@ -0,0 +1,31 @@
+                    {DepName, ".*", {git, Repo, {tag, Tag}}};
+               ({DepName, {git, Repo, {branch, Branch}}}) ->
+                    {DepName, ".*", {git, Repo, {branch, Branch}}};
+               ({HexDepName, HexDepVersion}) ->
```
do we need this case at all given the current `deps`?
Meaning this sound like a little over-engineering and over-generalization for questionable future changes.

okeuday replied:
rebar2 needs proper and covertool as normal deps, since it didn't have a concept of profiles.

waisbrot replied:
The objection, I assume, is to `({HexDepName, HexDepVersion})`?

I'd rather keep it because it's copy/paste from other projects, and if `proper` gets into Hex in a non-weird way (right now I think the Git and Hex releases don't match) then I'd want to swap the Hex link in.

motobob replied:
yes, that's the objection but not that a blocking one

motobob commented:
going to merge this and tag`1.2.2`

motobob commented:
oh...i see it's a `development` branch and not `master`. @waisbrot please merge/FF to master to it and tag it so to be used in hex and finally in erlcloud.

waisbrot commented:
I made `master` the main branch and deleted `development`.

I made the 1.2.2 tag and put it on Hex. Trying to fetch it into erlcoud, it looks OK to me (it doesn't grab the test deps).

I've added @motobob to the Hex package owner list.
